sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -280,6 +295,23 @@ def tune(
                 + exec_script
             )
 
+        resources = client.V1ResourceRequirements()
+        if isinstance(resources_per_trial, dict):
+            requests = {
+                "cpu": "200m",
+                "memory": "256Mi",
+            }
+            if "gpu" in resources_per_trial:
+                resources_per_trial["nvidia.com/gpu"] = resources_per_trial.pop("gpu")
+            requests.update(resources_per_trial)
+
+            resources = client.V1ResourceRequirements(
+                requests=requests,
+                limits=requests,
+            )
```

Comment:
I think, you can remove `requests` and only add `limits`.
Requests are not necessary and users always can use V1ResourceRequirements if they need it.

Reply:
WDYT @johnugeorge @tenzen-y @droctothorpe ?

Reply:
Let me preface by saying that if any of this is stating the obvious, I humbly apologize.

Requests **guarantee** resources. Limits do not. If you don't specify requests, there's no guarantee that you'll receive the required resources.

Here's an analogy to clarify the distinction.

Let's say that you make a dinner reservation for 4 at your favorite restaurant. Those 4 seats are now guaranteed.

If you only made reservations for 2 and arrived with 4, there's no guarantee that the restaurant can provide 4 seats.

Requests are like restaurant reservations in that sense.

Now let's say that you reserved 4 seats and 4 more friends decide to join last minute.

When you arrive at the restaurant with a total of 8 people, the restaurant may or may not be able to accommodate the extra 4. It depends on how many open seats there are. The restaurant is like a Kubernetes node in this analogy. Nodes often have unused resources, and if pods don't set limits or have limits that exceed requests, the Kubernetes scheduler will let a pod consume the un-utilized resources, much like your 4 extra friends taking up 4 extra seats in the restaurant. 

Limits prevent this behavior. If a restaurant is really crowded, they might tell your 4 extra friends to get lost. Limits don't guarantee a reservation, they just restrict a pod from consuming additional, unreserved resources beyond a specified threshold.

One of the benefit of not specifying limits is that it can improve node resource utilization. One of the drawbacks is that it makes your executions less consistent: performance varies based on how much unused resources are available on the cluster.

All of which is to say that, IMO, requests are much more important than limits, and it's probably advisable to have them match in this context -- they actually match by default if resources are not specified. A user who wants to disable limits can always just use `V1ResourceRequirements`.

Reply:
Great analogy @droctothorpe!
In general, I agree with you.
It seems that looking at this doc, if `requests` are not set, Kubernetes make `request=limits`: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits
But in that case, maybe it worth to specify `limits` and `requests` together to not confuse user.

Reply:
> If you specify a limit for a resource, but do not specify any request, and no admission-time mechanism has applied a default request for that resource, then Kubernetes copies the limit you specified and uses it as the requested value for the resource.
TIL!